engine/src/mbliphonedc.mm

```diff
@@ -859,16 +859,17 @@ static void MCScreenDCDoSnapshot(void *p_env)
 
 void MCScreenDC::getsystemappearance(MCSystemAppearance &r_appearance)
```

Comment:
I take it `__IPHONE_12_0` is defined if it is the iOS13 SDK too?

It would be better to ifdef the whole method.

Also I don't think you need respondsToSelector with the ifdef.

Reply:
@runrevmark yes each new SDK defines one of these for this purpose and we have used these elsewhere in the engine. Don't we need `respondsToSelector` in the event the engine compiled against > iOS 12 is run on older OS?

What do you mean by `ifdef` the whole method. A completely separate `getsystemappearance` implementation?

Reply:
Doh! Of course... We can actually get away without the ifdef at all:
```
typedef int (*_userInterfaceStyle)(UITraitCollection *, SEL selector);
UITraitCollection *t_traits = [MCIPhoneGetRootView() traitCollection];
if ([t_traits respondsToSelector: @selector(userInterfaceStyle)] &&
        ((_userInterfaceStyle)objc_msgSend)(t_traits, @selector(userInterfaceStyle)) == 2)
...
```